### PR TITLE
Adds support for GitLab project Freeze Periods API

### PR DESCRIPTION
**Pull request by pcnc**
This PR implements the freeze periods API added for projects in GitLab 13.0. 

https://docs.gitlab.com/ee/api/freeze_periods.html

### CODE REVIEW

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Could you make a new service for this called `FreezePeriodsService` and put all the code in a decidated file called `freeze_periods.go`? That would make this new code (API) be inline with how the rest of the package is structured.

Once that I done, I'll review the code itself. But on first glance that looks pretty well...

**Comment by pcnc** [comment]
Refactored all new code under the `FreezePeriodsService` service, located in a separate service-specific file

File: freeze_periods.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/freeze_periods", pathEscape(project))
+
+	req, err := s.client.NewRequest("GET", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var pfp []*FreezePeriod
```
Comment (svanharmelen) [inline freeze_periods.go:66]:
```suggestion
	var fps []*FreezePeriod
```

File: freeze_periods.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/freeze_periods/%d", pathEscape(project), freezePeriod)
+
+	req, err := s.client.NewRequest("GET", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pfp := new(FreezePeriod)
```
Comment (svanharmelen) [inline freeze_periods.go:91]:
```suggestion
	fp := new(FreezePeriod)
```

File: freeze_periods.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+	resp, err := s.client.Do(req, pfp)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return pfp, resp, err
+}
+
+// CreateFreezePeriodOptions represents the available CreateFreezePeriodOptions() options.
```
Comment (svanharmelen) [inline freeze_periods.go:100]:
```suggestion
// CreateFreezePeriodOptions represents the available CreateFreezePeriodOptions()
// options.
```

File: freeze_periods.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/freeze_periods.html#create-a-freeze-period
+type CreateFreezePeriodOptions struct {
+	FreezeStart  *string `json:"freeze_start"`
+	FreezeEnd    *string `json:"freeze_end"`
+	CronTimezone *string `json:"cron_timezone,omitempty"`
```
Comment (svanharmelen) [inline freeze_periods.go:107]: Please add `url` tags and `omitempty` to all fields here...

File: freeze_periods.go [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/freeze_periods", pathEscape(project))
+
+	req, err := s.client.NewRequest("POST", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pfp := new(FreezePeriod)
```
Comment (svanharmelen) [inline freeze_periods.go:126]:
```suggestion
	fp := new(FreezePeriod)
```

File: freeze_periods.go [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+// UpdateFreezePeriodOptions represents the available UpdateFreezePeriodOptions() options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/freeze_periods.html#update-a-freeze-period
+type UpdateFreezePeriodOptions struct {
+	FreezeStart  *string `json:"freeze_start,omitempty"`
+	FreezeEnd    *string `json:"freeze_end,omitempty"`
+	CronTimezone *string `json:"cron_timezone,omitempty"`
```
Comment (svanharmelen) [inline freeze_periods.go:142]: Please add `url` tags...

File: freeze_periods.go [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/freeze_periods/%d", pathEscape(project), freezePeriod)
+
+	req, err := s.client.NewRequest("PUT", u, opt, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pfp := new(FreezePeriod)
```
Comment (svanharmelen) [inline freeze_periods.go:161]:
```suggestion
	fp := new(FreezePeriod)
```

File: freeze_periods.go [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+	}
+	u := fmt.Sprintf("projects/%s/freeze_periods/%d", pathEscape(project), freezePeriod)
+
+	req, err := s.client.NewRequest("DELETE", u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
```
Comment (svanharmelen) [inline freeze_periods.go:188]: Please end the file with a newline.

File: freeze_periods.go [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+	resp, err := s.client.Do(req, pfp)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return pfp, resp, err
+}
+
+// DeleteFreezePeriod removes a freeze period from a project. This is an idempotent
+// method and can be called multiple times. Either the hook is available or not.
```
Comment (svanharmelen) [inline freeze_periods.go:171]:
```suggestion
// DeleteFreezePeriod removes a freeze period from a project. This is an
// idempotent method and can be called multiple times. Either the hook is
// available or not.
```

File: freeze_periods.go [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+	pfp := new(FreezePeriod)
+	resp, err := s.client.Do(req, pfp)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return pfp, resp, err
+}
+
+// UpdateFreezePeriodOptions represents the available UpdateFreezePeriodOptions() options.
```
Comment (svanharmelen) [inline freeze_periods.go:135]:
```suggestion
// UpdateFreezePeriodOptions represents the available UpdateFreezePeriodOptions()
// options.
```

**Review by svanharmelen — COMMENTED** [review]
Only very minor comments which are easy to fix. Rest looks good 👍🏻

**Comment by pcnc** [comment]
Thanks for the suggestions! All requested changes were implemented.

**Review by svanharmelen — APPROVED** [review]
Super nice, thanks @pcnc!

**Pull request merged**